elfpy/utils/apeworx_integrations.py

```diff
@@ -35,13 +35,24 @@ def get_transfer_single_event(tx_receipt: Receipt) -> ContractLog:
         <https://docs.apeworx.io/ape/stable/methoddocs/types.html#ape.types.ContractLog>`_
         ) event, excluding periferal events.
     """
-    single_events: list[ContractLog] = []
-    for tx_event in tx_receipt.events:
-        if tx_event.event_name == "TransferSingle":
-            single_events.append(tx_event)
-    if len(single_events) != 1:
-        raise ValueError(f'The transaction receipt should have one "TransferSingle" event, not {len(single_events)}.')
-    return single_events[0]
```

Comment:
I think the new logic avoids the undesired condition of multiple single events (`!=1`). You'll want to keep that one in, we don't want to just automatically return the [-1] index.

Reply:
added another filter to exclude id=0 events only when we have more than one to choose from. this seems safe, since id is the field we're decoding, so we want the entry that's non-empty, if possible. don't see any edge cases anymore. diff here: https://github.com/element-fi/elf-simulations/pull/321/commits/9b41e095e44069907b4606a1efc9d0f2871a21d3